src/Tasks/GenerateManifestBase.cs

```diff
@@ -619,7 +619,13 @@ private bool WriteManifest()
             }
             catch (Exception ex)
             {
-                Log.LogErrorWithCodeFromResources("GenerateManifest.WriteOutputManifestFailed", OutputManifest.ItemSpec, ex.Message);
+                string lockedFileMessage = string.Empty;
+                if (NativeMethodsShared.IsWindows)
+                {
+                    lockedFileMessage = LockCheck.GetLockedFileMessage(OutputManifest.ItemSpec);
+                }
```

Comment:
nit:
```suggestion
                string lockedFileMessage = LockCheck.GetLockedFileMessage(OutputManifest.ItemSpec);
```

(The IsWindows check is built into GetLockedFileMessage.)

Reply:
If have no this condition, it will haveerror CA1416: This call site is reachable on all platforms. 'LockCheck.GetLockedFileMessage(string)' is only supported on: 'windows'. (h 
ttps://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1416)

Reply:
I didn't notice there was a SupportedOSPlatform check around all of LockCheck. In that case, line 258 of LockCheck doesn't need the NativeMethodsShared.IsWindows check.

Reply:
Removed the duplicate one.

Reply:
Thanks!